### DISCUSSION TITLE
Does the library support multi-layer EXR files?

### DISCUSSION THREAD

**Discussion by vanka78bg** [dlemstra/Magick.NET#1608] [Help]
Hello, all!

I have an EXR file which has multiple layers. I am trying to open it and read the layers one-by-one like this:

```csharp
using var images = new MagickImageCollection("some_file.exr");

foreach (var image in images)
{
    // TODO: Process the image layer
}
```

The result is that I receive only the first layer, containing the RGBA color data. The other layers are not returned, i.e. the collection contains only a single image.

I have also tried to create a multi-layer image myself:

```
using var images = new MagickImageCollection();
using var image1 = new MagickImage("image1.png");
using var image2 = new MagickImage("image2.png");
using var image3 = new MagickImage("image3.png");

images.Add(image1);
images.Add(image2);
images.Add(image3);
images.Write("image.tiff");
images.Write("image.exr");
```

The above code tries to create a single multi-layer image form multiple PNG files. This produces a single TIFF file with multiple pages, which is expected. Unfortunately, the EXR output consists of multiple files: image-0.exr, image-1.exr, image-2.exr, etc.

 What am I doing wrong?

**Comment by dlemstra** [comment]
It looks like the ImageMagick currently has no support for reading multiple layers in an exr file. Is it possible to share a file so I can see if it would be possible to add support for this?
